src/Stitch.hs

```diff
@@ -20,20 +21,19 @@ createArguments introFileName outputDirectory mainFilePath =
       outputFileName = outputDirectory <> format fp (Turtle.filename mainFilePath)
   in ["-i", introFileName, "-i", mainFileName, "-filter_complex", "concat=n=2:v=0:a=1", outputFileName]
 
-getFilesAndPaths :: Reader Config (Turtle.Text, Turtle.Text, Turtle.Text)
-getFilesAndPaths = do
-  inputDirectory <- asks (lookUpConfig "EDIT_INPUT_DIRECTORY")
-  outputDirectory <- asks (lookUpConfig "EDIT_OUTPUT_DIRECTORY")
-  editingTemplateDirectory <- asks (lookUpConfig "EDITING_TEMPLATE_DIRECTORY")
-  introFileName <- asks (lookUpConfig "INTRO_FILENAME")
-  return (inputDirectory, outputDirectory, editingTemplateDirectory <> introFileName)
-
 runCommand :: Turtle.Text -> Turtle.Text -> Turtle.FilePath -> IO ()
 runCommand introFileName outputDirectory filePath = proc "ffmpeg" (createArguments introFileName outputDirectory filePath) Turtle.empty >>= print
 
-stitch :: IO ()
-stitch = let
-  (inputDirectory, outputDirectory, introFileName) = runReader getFilesAndPaths appConfig
-  in
-    foldIO (getMP3FilesFromPath inputDirectory) (L.sink (runCommand introFileName outputDirectory))
+appStitch :: ReaderT Config IO ()
+appStitch = do
+  eitherInputDirectory <- asks (lookUpConfig "EDIT_INPUT_DIRECTORY")
+  eitherOutputDirectory <- asks (lookUpConfig "EDIT_OUTPUT_DIRECTORY")
+  eitherTemplateDirectory <- asks (lookUpConfig "EDITING_TEMPLATE_DIRECTORY")
+  eitherIntroFilename <- asks (lookUpConfig "INTRO_FILENAME")
```

Comment:
I really should just provide the filename with the full path and avoid combining them here.